### PR TITLE
Aggregate missing team and user approvals from policy evaluation results

### PR DESCRIPTION
**Pull request by gjonathanhong**

### CODE REVIEW

File: pkg/commands/policy/policy_test.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			c := &PolicyCommand{
+				results: tc.results,
+			}
+
+			err := c.Process(ctx)
+			if diff := testutil.DiffErrString(err, tc.wantErr); diff != "" {
```
Comment (gjonathanhong) [inline pkg/commands/policy/policy_test.go:99]: question for reviewer: do i need to cuddle the err here if I'm already cuddling diff?

File: pkg/commands/policy/policy.go [hunk 2]
```diff
@@ -90,5 +96,40 @@ func (c *PolicyCommand) Run(ctx context.Context, args []string) error {
...
+
+		if len(v.MissingApprovals) == 0 {
+			logger.DebugContext(ctx, "no missing approvals for policy",
+				"policy_name", k)
+			continue
+		}
+
+		for _, m := range v.MissingApprovals {
+			c.teams = append(c.teams, m.AssignTeams...)
+			c.users = append(c.users, m.AssignUsers...)
```
Comment (verbanicm) [inline pkg/commands/policy/policy.go:122]: This seems unnecessary and is only used to try to get the test results you want. I would just test the response message. When you add the functionality to assign reviewer, mock the GitHub client and expect the requests sent contain these reviewers instead (similiar to how plan and apply are tested).

File: pkg/commands/policy/policy.go [hunk 3]
```diff
@@ -90,5 +96,40 @@ func (c *PolicyCommand) Run(ctx context.Context, args []string) error {
...
 
-	return nil
+	c.results = results
+
+	return c.Process(ctx)
+}
+
+// Process handles the main logic for handling the results of the policy
+// evaluation.
+func (c *PolicyCommand) Process(ctx context.Context) error {
```
Comment (verbanicm) [inline pkg/commands/policy/policy.go:106]: You could accept the results as a param here or move the code process the results file into the process function and use a file from a testdata folder inside this dir to test the functionality fully.

File: pkg/commands/policy/policy_test.go (hunk 1 above)
Comment (verbanicm) [reply pkg/commands/policy/policy_test.go:100]: I believe this is correct.

File: pkg/commands/policy/policy.go (hunk 2 above)
Comment (gjonathanhong) [reply pkg/commands/policy/policy.go]: Removed from struct and moved all logic (except for parsing flags) into `Process`

File: pkg/commands/policy/policy.go (hunk 3 above)
Comment (gjonathanhong) [reply pkg/commands/policy/policy.go]: Opted for latter

File: pkg/commands/policy/testdata/results.json
Comment (verbanicm) [inline pkg/commands/policy/testdata/results.json:1]: Is this file used?

**Review by verbanicm — APPROVED** [review]

**Review by sethvargo — APPROVED** [review]

File: pkg/commands/policy/testdata/results.json
Comment (gjonathanhong) [reply pkg/commands/policy/testdata/results.json:1]: 💥 💣💥 gone 💥 💣💥

  😁

**Pull request merged**